### DISCUSSION TITLE
Subscription to node properties changes

### DISCUSSION THREAD

**Discussion by alejandro-zuleta-globant** [FreeOpcUa/opcua-asyncio#1141] [Q&A] [answered]
I have a server whose address space is populated with a couple of `UAObject` objects, these objects have multiple `UAVariables` properties using the has `HasProperty` reference.

Based on the examples, I think I can use `Subscription.subscribe_data_change` but I have to pass all the property nodes to watch. I would like to simplify the subscription process and just pass the parent object and get notified if any property of the object is changed.

Is there a way to subscribe to an object node and get notified whenever _**any**_ property value related to the object is modified?

**Comment by AndreasHeine** [comment, marked as answer]
thats not how a OPC UA Subscription (more precise a MonitoredItem) works!

if you want to subscribe all properties of an object you need to browse the object for the references (HasProperty) and pass the a list of nodeids to the subscribe_data_change() method

you can wrap that in your own conveniencefunction and use it as you wish!

**Reply by alejandro-zuleta-globant** [reply]
Got it, I was getting this wrong. Thank you for your reply.
